frontend/src/Frontend/KadenaAddress.hs

```diff
@@ -107,11 +109,12 @@ data KadenaAddressPrefix
   | KadenaAddressPrefix_None
   deriving (Show, Eq)
 
-data AccountCreated
-  = AccountCreated_Yes
-  | AccountCreated_No
+newtype AccountCreated = AccountCreated Bool
   deriving (Show, Eq)
 
+accountCreatedBool :: a -> a -> AccountCreated -> a
+accountCreatedBool no yes (AccountCreated b) = bool no yes b
```

Comment:
Boolean blindness here is terrible, I'd rather just pattern match anywhere this is used